core/src/test/java/io/questdb/test/griffin/SqlOptimiserTest.java

```diff
@@ -4082,6 +4082,32 @@ public void testSampleByGroupByFillNone() throws Exception {
         });
     }
 
+    @Test
+    public void testSampleByWithDesignedTimestamp() throws Exception {
+        execute("CREATE TABLE 't' (\n" +
+                "  name SYMBOL capacity 256 CACHE,\n" +
+                "  price DOUBLE,\n" +
+                "  timestamp TIMESTAMP\n" +
+                ") timestamp (timestamp) PARTITION BY DAY;");
+        execute("INSERT INTO t (name, price, timestamp) VALUES" +
+                " ('a', 10, '2023-09-01T00:00:00.000Z')," +
+                " ('a', 11, '2023-09-01T00:10:00.000Z')");
+
+        assertMemoryLeak(() -> {
+            final String query = "SELECT " +
+                    "min(price) AS min_price, max(price) AS max_price, timestamp (timestamp + 1000000) AS new_ts " +
```

Comment:
What does `timestamp (timestamp + 1000000)` mean? We don't have `timestamp` SQL function and it doesn't look like supported cast syntax.

Reply:
I'm modifying the sample by rewrite ATM in #5493 and I'd like to include a fix into that PR. Is that ok?

Reply:
> Yes, it doesn't make sense for our db. I'm just confirming whether an expression can be used as an argument for this function. Perhaps it would be better if the argument were an expression rather than a literal column?

That SQL rewrite code is too optimistic: it doesn't check the AST node type and simply check the root node's name. So, if you have a timestamp column named, for instance, `to_utc`, the code will break. There is more buggy code there, so I'll add more test cases and include a fix for this bug into #5493 - the query will return a meaningful error instead of throwing index out of bounds exception.